xcm/xcm-executor/src/lib.rs

```diff
@@ -60,7 +65,7 @@ impl<Config: config::Config> ExecuteXcm<Config::Call> for XcmExecutor<Config> {
 			return Outcome::Error(XcmError::WeightLimitReached(maximum_weight));
 		}
 		let mut trader = Config::Trader::new();
-		let result = Self::do_execute_xcm(origin, true, message, &mut 0, Some(shallow_weight), &mut trader);
+		let result = Self::do_execute_xcm(origin, true, message, &mut weight_credit, Some(shallow_weight), &mut trader);
```

Comment:
Can you elaborate on why these need to be mutable?

Reply:
Not sure what you mean. This API already existed.